dali/pipeline/operators/operator.h

```diff
@@ -256,6 +256,26 @@ void CollectPointersForExecution(size_t batch_size,
                                  const TensorList<GPUBackend> &input, vector<const uint8 *> *inPtrs,
                                  TensorList<GPUBackend> *output, vector<uint8 *> *outPtrs);
 
+// Macros for writing RunImpl templated code for derived operators
+#define RUN_IMPLEMENT(ws, idx, F16C)        \
+     DataDependentSetup(ws, idx);           \
+     if (output_type_ == DALI_FLOAT16)      \
+        F16C(ws, idx);                      \
+     else                                   \
+        DALI_IMAGE_TYPE_SWITCH_NO_FLOAT16(  \
+            output_type_, imgType,          \
+            RunHelper<imgType>(ws, idx))
+
+#define RUN_IMPLEM(ws, idx, half)  RUN_IMPLEMENT(ws, idx, RunHelper<half>)
+#define RUN_IMPL_GPU(ws, idx)      RUN_IMPLEM(ws, idx, float16)
+
+#ifdef DALI_F16C
+  #define RUN_IMPL_CPU(ws, idx)    RUN_IMPLEMENT(ws, idx, RunHelperF16C)
+  #define CVTSS_SH(x)              _cvtss_sh(x, 0)
```

Comment:
Why we need to use this macro? Hiding that as a specialized code path inside half.hpp won't work?

Reply:
And if we going to use more intrinsic having them macro defined is a bad idea.